File: languages/go/entity/identityV2/decimal_attribute.go
package identityV2

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// DecimalAttribute represents an attribute with a decimal value.
type DecimalAttribute struct {
	Attribute
}

// NewDecimalAttribute creates a new DecimalAttribute instance with the provided key and value.
func NewDecimalAttribute(key string, value float64) DecimalAttribute {
	a := Attribute{
		Id:    key,
		Value: value,
	}
	return DecimalAttribute{
		Attribute: a,
	}
}

func NewDecimalAttributeFromProto(s *proto.DecimalAttributeV2) DecimalAttribute {
	if s == nil {
		return DecimalAttribute{}
	}
	a := Attribute{
		Id:    s.GetId(),
		Value: s.GetValue(),
	}
	return DecimalAttribute{
		Attribute: a,
	}
}

func (s DecimalAttribute) ToProto() *proto.DecimalAttributeV2 {
	return &proto.DecimalAttributeV2{
		Id:    s.Attribute.Id,
		Value: s.Attribute.Value.(float64),
	}
}

File: languages/java/src/main/java/com/bloock/sdk/entity/identity/DecimalAttribute.java
package com.bloock.sdk.entity.identity;

import com.bloock.sdk.bridge.proto.BloockIdentityEntities;

/**
 * Represents an attribute with a decimal value.
 */
public class DecimalAttribute extends Attribute<Double> {
  /**
   * Creates a new DecimalAttribute instance with the provided key and value.
   * 
   * @param id
   * @param value
   */
  public DecimalAttribute(String id, double value) {
    super(id, value);
  }

  public static DecimalAttribute fromProto(BloockIdentityEntities.DecimalAttribute res) {
    return new DecimalAttribute(res.getId(), res.getValue());
  }

  public BloockIdentityEntities.DecimalAttribute toProto() {
    return BloockIdentityEntities.DecimalAttribute.newBuilder()
        .setId(this.id)
        .setValue(this.value)
        .build();
  }
}
